### ISSUE TITLE
Behaviour when no host is specified in swagger file

### ISSUE THREAD

**Issue by gwandwingor** [Axway-API-Management-Plus/apim-cli#53]
Hi,

I wonder how does the apim-cli behaves when no "host" field is provided in a swagger file ?

I recently found that in apimanager-swagger-promote v1.6.7-1, when such a thing occurred, the following error was thrown but the API was still successfully imported and backend host was set to "http://", which doesn't help since the Frontend API will not be able to reach any backend.

Would such a behaviour be the same in apim-cli, I think the most useful thing to do would be to : 

- if no "backendBasepath" is specified in the config file : end the API processing with an error status so the problem can be addressed straight away.
- if "backendBasepath" is specified in the config file : create the "host" entry in the swagger file and set it to the "backendBasepath" host

log example with swagger-promote :
```
------------------------------------------------------------------------
API-Manager Promote: 1.6.7-1 - I M P O R T
                                                                        
To report issues or get help, please visit: 
https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote
------------------------------------------------------------------------

0      INFO  nvironmentProperties| Trying to load environment properties from file: env.properties ... not found.
0      INFO  nvironmentProperties| Trying to load environment properties from file: env.DEV.properties ... not found.
663    WARN  sponseProcessCookies| Invalid cookie header: "Set-Cookie: SSOSPAPIMAN=deleted;Version=1;Comment=;Domain=;Path=/api/portal/v1.3/;Max-Age=0;Secure;HttpOnly". Blank or null value for domain attribute
1010   INFO  IImportConfigAdapter| Loaded stage API-Config from file: /home/jenkins/agent/workspace/livery-apimgr-mobileoperator_DEV/download/api/apimgr-mobileoperator/common-api/config/common-config.DEV.json
1109   INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: '/home/jenkins/agent/workspace/livery-apimgr-mobileoperator_DEV/download/api/apimgr-mobileoperator/common-api/swagger/common-api.json' (relative path)
1127   ERROR     APISpecification| Cannot replace host in provided Swagger-File. Continue with given host.
java.lang.NullPointerException
	at com.axway.apim.api.definition.Swagger2xSpecification.configureBasepath(Swagger2xSpecification.java:55)
	at com.axway.apim.api.definition.Swagger2xSpecification.configure(Swagger2xSpecification.java:93)
	at com.axway.apim.api.definition.APISpecificationFactory.getAPISpecification(APISpecificationFactory.java:41)
	at com.axway.apim.apiimport.APIImportConfigAdapter.getDesiredAPI(APIImportConfigAdapter.java:218)
	at com.axway.apim.App.run(App.java:227)
	at com.axway.apim.App.main(App.java:53)
1264   INFO  IImportConfigAdapter| Handling configured client-applications.
1405   INFO  IImportConfigAdapter| Found existing application: 'MOBILE-SAFETY-MOB-REACT-DEV' (null) based on given name 'MOBILE-SAFETY-MOB-REACT-DEV'
1405   INFO  IImportConfigAdapter| Found existing application: 'MOBILE-SAFETY DEV' (null) based on given name 'MOBILE-SAFETY DEV'
1442   INFO     APIManagerAdapter| API-Manager version is: 7.5.3 SP12
1443   INFO               Proxies| Sending request to find existing APIs: https://apim-group-dev.tools.endpoint:8075/api/portal/v1.3/proxies
1677   INFO               Proxies| No existing API found based on filter: [apiPath=/dev/mobile-operator/common filter=[]]
1681   INFO      APIImportManager| Strategy: No existing API found, creating new!
1684   INFO     APIManagerAdapter| Importing backend API (Swagger/WSDL Import)
1811   INFO     APIManagerAdapter| Create API-Proxy (Front-End API)
5210   INFO     APIManagerAdapter| Updating Frontend-API (Proxy) for the following properties: caCerts securityProfiles tags corsProfiles path name summary 
5391   INFO     APIManagerAdapter| Creating API-Access for the following apps: '[[MOBILE-SAFETY-MOB-REACT-DEV], [MOBILE-SAFETY DEV]]'
5479   INFO                   App| Successfully replicated API-State into API-Manager 
```

**Comment by cwiechmann** [comment]
Yes, I think the behavior in the APIM-CLI is the same and by far not perfect.
Your recommendations make sense and I’ll change the behavior in the APIM-CLI only to support the scenario you described.

**Comment by cwiechmann** [comment]
This issue has been partly fixed in version 1.2.3. If no host is given, there should be no NPE exception anymore. 

However, it still makes sense to stop processing, if no host is present and the backendBasepath is also not set.

**Comment by cwiechmann** [comment]
With version 1.2.4, when replicating/importing an API without having a host and backendBasepath it will be stopped.
